src/Control/Exception/Annotated.hs

```diff
@@ -224,17 +234,23 @@ try action =
       `catch`
           (\exn -> pure $ Left exn)
 
--- | Attaches the 'CallStack' to the 'AnnotatedException' that is thrown.
+-- | Throws an 'Exception' and annotates it with the current 'CallStack'.
 --
--- The 'CallStack' will *not* be present as a 'CallStack' - it will be
--- a 'CallStackAnnotation'.
+-- An alias for 'throwWithCallStack'.
```

Comment:
This looks weird to me. Is there really no difference between `throw` and `throwWithCallStack`? Their implementations are different. If they do the same thing in practice, should one actually be an alias of the other, or should we just remove one of them?

Reply:
The only difference should be in `withFrozenCallStack` which prevenst the callstack from having `["throw", "throwWIthCallStack"]`.

Reply:
How should I know when to use one vs the other?

Reply:
Everyone: "Haskell code has no side effects."

HasCallStack: "Hold my beer."

Reply:
I suppose we could drop one of them. I like having the alias around to keep `Control.Exception.Annotated` (as much as possible) a drop-in replacement for other `Control.Exception` code. 

Really, though, there's a lot that needs to be done there to make the APIs totally compatble.

Kinda wish `backpack` were easier to use, seems like a natural desire 🤔

Reply:
I don't know that I fully agree with this idea of giving the functions the same name as some functions in some other module as a "drop in" replacement. It's almost like "reassignment by shadowing" or like dynamic scoping of variables ("extreme late binding in all things," right?). It just seems like a very error-prone practice, and all it really saves you from is having to do a regex replace.

Reply:
Ah, of course. Keeping it around just to make it slightly easier to switch over makes sense to me, even if it’s not currently a full drop-in replacement.